torch/utils/data/sampler.py

```diff
@@ -117,7 +117,7 @@ def __init__(self, sampler, batch_size, drop_last):
     def __iter__(self):
         batch = []
         for idx in self.sampler:
-            batch.append(idx)
+            batch.append(int(idx))
```

Comment:
Any particular reason to cast index to integer? It is never documented that the batch sampler accepts only number-like indexes. It was quite handy to pass an object to index a complicated dataset.

Reply:
The issue was that RandomSampler was using `LongTensors` as an index to the dataset. The Variable and Tensor merge had the effect that indexing a 1-d LongTensor return a 0-d (scalar) LongTensor instead of a number. The 0-d tensor was much slower to serialize.

The other part of the fix was this line:
https://github.com/pytorch/pytorch/pull/5225/files/e22441c7d8edd20473033ec8d50e657604b33d02#diff-b8821844d3c39c6b229d352972c72019R50

The change to RandomSampler is probably sufficient. We could probably remove this `int()` call.

Reply:
Actually, this is addressed in #8652